### PR TITLE
EVM RPC canister documentation 

### PR DESCRIPTION
**Pull request by jessiemongeon1**
Thank you for your contribution to the IC Developer Portal.
Before submitting your Pull Request, please make sure that:

- [ ] You have updated the [`.github/CODEOWNERS`](https://github.com/dfinity/portal/blob/master/.github/CODEOWNERS) file according to your changes.
- [ ] You have registered new documents to [`/sidebars.js`](https://github.com/dfinity/portal/blob/master/sidebars.js) if any.

### CODE REVIEW

File: docs/developer-docs/integrations/ethereum/evm-rpc.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+- `eth_getBlockByNumber`: Queries information about a given block.
+- `eth_getTransactionReceipt`: Queries details about a submitted transaction.
+- `eth_feeHistory`: Queries the historical fee data to estimate gas prices for transactions.
+- `eth_sendRawTransaction`: Submits a signed transaction to the Ethereum network.
+- `eth_getTransactionCount`: Queries the number of transactions for a specified address.
+- `eth_call`: Calls a smart contract function without changing the blockchain's state.
+- `eth_getStorageAt`: Queries the value of a storage position at a given address.
+- `eth_estimateGas`: Provides an estimate for the gas consumption of a transaction before it is mined.
+- `eth_blockNumber`: Queries the number of the most recent block.
+- `eth_getCode`: Queries the bytecode of a specified contract.
```
Comment (rvanasa) [inline docs/developer-docs/integrations/ethereum/evm-rpc.md:35]: We currently only have first-class support for six of these methods, while the rest can be accessed with the `request` method. Here's a suggestion to give a sense for the distinction:

```suggestion
- `eth_feeHistory`: Queries the historical fee data to estimate gas prices for transactions.
- `eth_getLogs`: Queries the logs of a specified block or transaction.
- `eth_getBlockByNumber`: Queries information about a given block.
- `eth_getTransactionCount`: Queries the number of transactions for a specified address.
- `eth_getTransactionReceipt`: Queries details about a submitted transaction.
- `eth_sendRawTransaction`: Submits a signed transaction to the Ethereum network.

Other JSON-RPC methods (including those specific to non-Ethereum networks) may be accessed using the canister's `request` method.
```

File: docs/developer-docs/integrations/ethereum/evm-rpc.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+- Ankr: `https://rpc.ankr.com/eth/`
+- BlockPI: `https://ethereum.blockpi.network/v1/rpc/`
+- Cloudflare Web3: `https://cloudflare-eth.com/v1/mainnet/`
+- Public Node: `https://ethereum.publicnode.com`
+
+View the [full list of RPC providers for each EVM network.](https://chainlist.org/)
+
+### Specifying an RPC provider
+
+To specify the RPC provider you'd like to use, change the `Url` value:
```
Comment (rvanasa) [inline docs/developer-docs/integrations/ethereum/evm-rpc.md:50]: It's also possible to specify RPC providers in other ways. Here's the Candid interface:

```candid
type JsonRpcSource = variant {
  Chain : nat64;
  Service : record { hostname : text; chainId : opt nat64, headers: opt vec (text, text) };
  Provider : nat64;
  Custom : record { url : text; headers : vec (text, text) };
};

request : (
  source : JsonRpcSource,
  jsonRequest : text,
  maxResponseBytes : nat64
) -> (
  Result<text, RpcError>
);
```

I suppose we could directly include this in the documentation as a reference point for various use cases.

File: docs/developer-docs/integrations/ethereum/evm-rpc.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+```
+dfx canister call evm_rpc --wallet $(dfx identity get-wallet) --with-cycles 600000000 request '(variant {Url="https://ethereum.publicnode.com"},"{\"jsonrpc\":\"2.0\",\"method\":\"eth_gasPrice\",\"params\":[],\"id\":1}",1000)'
+```
+
+### Registering your own provider
+
+To register your own RPC provider, you can use the following command:
+
+```
+dfx canister call evm_rpc register_provider '(record { chain_id=1; base_url="https://cloudflare-eth.com"; credential_path="/v1/mainnet"; cycles_per_call=10; cycles_per_message_byte=1; })'
```
Comment (rvanasa) [inline docs/developer-docs/integrations/ethereum/evm-rpc.md:61]: These examples on the readme were a bit outdated; I'll go through and double-check / update these in a PR which merges into this one.

File: docs/developer-docs/integrations/ethereum/evm-rpc.md (hunk 2 above)
Comment (jessiemongeon1) [reply docs/developer-docs/integrations/ethereum/evm-rpc.md:46]: Added this to the doc

File: docs/developer-docs/integrations/ethereum/evm-rpc.md (hunk 3 above)
Comment (rvanasa) [reply docs/developer-docs/integrations/ethereum/evm-rpc.md:76]: PR: https://github.com/dfinity/portal/pull/2413

File: docs/developer-docs/integrations/ethereum/overview.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Costs
+
+An approximate cost breakdown in USD can be found below. These costs are estimated using the Cloudflare RPC with a 13-node subnet:
+
+- JSON-RPC API request: $0.0001 USD
+- HTTPS outcalls: $0.001 USD (This assumes 1KiB request and 1KiB response)
+- Base cost: $0.0008 USD
+- JSON Request: $0.00008 USD per KiB
+- JSON Response: $0.00008 USD per KiB
```
Comment (rvanasa) [inline docs/developer-docs/integrations/ethereum/overview.md:21]: Making a quick note that, at some point, we need to recompute these costs using the new HTTP outcall cost calculations and measured cost-per-compute-unit for the JSON-RPC APIs.

Comment (rvanasa) [reply docs/developer-docs/integrations/ethereum/overview.md]: @jessiemongeon1, I updated the "Costs" section based on the latest numbers. Feel free to adjust the wording, etc.

**Review by dfx-json — APPROVED** [review]

**Pull request merged**